Build fix with MinGW GCC 9.2.0

cristianadam opened the pull request:
**Describe the pull request**

This PR will allow building vcpkg itself with MinGW GCC 9.2.0. It fixes some warnings and because warnings are treated as errors, the build would fail.

It also fixes the usage of MinGW community tripplets.

Fixes: #12065

cristianadam committed:
Build fix with MinGW GCC 9.2.0

cristianadam committed:
Fix MinGW tripplets

Fixes #12065

NancyLi1013 commented:
/azp run

JackBoosY commented:
/azp run

strega-nil commented on toolsrc/src/vcpkg/build.cpp:
```diff
@@ -450,7 +450,7 @@ namespace vcpkg::Build
         Checks::check_exit(VCPKG_LINE_INFO, out_file, "Failed to open '%s' for writing", stdoutlog.u8string());
         std::string compiler_hash;
-        const int return_code = System::cmd_execute_and_stream_lines(
+        /*const int return_code = */System::cmd_execute_and_stream_lines(
```
This seems incorrect.

strega-nil reviewed (approved):
Can you merge to latest master? This seems like something we definitely want.

Commit:
clang-format

cristianadam committed:
Fix unused read_symlink_implementation warning marked as error

read_symlink_implementation was used only in copy_symlink_implementation
when #if defined(_WIN32) && !VCPKG_USE_STD_FILESYSTEM  was true.

Removed the warning otherwise.

cristianadam commented:
> Can you merge to latest master? This seems like something we definitely want.

I've rebased on master.
